shim/src/unsafeRec.js

```diff
@@ -31,7 +31,7 @@ function createNewUnsafeGlobalForNode() {
   return unsafeGlobal;
 }
 
-function createNewUnsafeGlobalForBrowser() {
+export function createNewUnsafeGlobalForBrowser() {
```

Comment:
you'd need a decorator for that ;-)

Reply:
@jfparadis this wouldn't be exclusive to tape; you'd use a decorator that made a private thing public (or made an exported `undefined` binding be the actual function, or something) when `process.env.NODE_ENV` was `'test'`.

Reply:
@ljharb, I agree, it's not exclusive to `tape`, and I'm wondering one of the many extensions could provide the behaviour we need.

The main issue: after we invoke our methods that repair the environment, using decorators or not, there is no complete revert, and tests can easily bleed into all subsequent calls. We could of course develop a "capture & restore" plugin for the global and its intrinsics, but it's much simpler to start with a new realm. This is an area where test262 is stronger, every test being executed in a new realm with a fresh set of intrinsics.

In my mind, it's more important to solve this issue with the execution environment, and doing so would automatically resolve all other issues: we simply load the class in two contexts, and then check the correct behaviour without export or decorators.

Reply:
it looks like `multi-tape` was designed to do what we need:

> Since it runs all tests in a single node instance, global variables and mocks will spread between the tests in an uncontrolled manner. `multi-tape` solves this by running each js-file in a separate node-process.
https://www.npmjs.com/package/multi-tape